xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
But see it the other way round: The thread could already be ended, but `IsRunning()` is still returning true. I really don't know if this could happen here, but this scenario is possible in an other context.

Regardless, you admitted that there is a data race. It maybe isn't relevant in the places where it is used today. But at some point someone could rely on the "correct" behaviour of `IsRunning()` and of `ThreadId()`. I think making `m_ThreadId` a std::atomic\<int> doesn't hurt and saves us from possible errors in the future.

Anyway, I think this discussion reached a dead end. If you still think there should be no synchronization, I remove this commit. The change can then be discussed in a separate  pull request if someone is interested in it.